module/x/gravity/keeper/batch.go

```diff
@@ -292,9 +292,9 @@ func (k Keeper) GetLastOutgoingBatchByTokenType(ctx sdk.Context, token types.Eth
 	batches := k.GetOutgoingTxBatches(ctx)
 	var lastBatch *types.InternalOutgoingTxBatch = nil
 	lastNonce := uint64(0)
-	for _, batch := range batches {
+	for i, batch := range batches {
 		if batch.TokenContract.GetAddress() == token.GetAddress() && batch.BatchNonce > lastNonce {
-			lastBatch = &batch
+			lastBatch = &batches[i]
```

Comment:
does this change have any practical effect? As far as I can tell both will return a pointer to the same location in memory. If there's some pointer related issue here it would be good to comment.

Reply:
Here the iteration variable batch will always have the same memory address, but its value will change in each iteration since for each entry  of an array range statement assigns iteration values to corresponding iteration variables.
Here there is an example of the same issue (they fix it in a little different way)
https://medium.com/swlh/use-pointer-of-for-range-loop-variable-in-go-3d3481f7ffc9

So if we have the batch array with nonce values {3, 2, 1}, the old code would return batch with nonce 1 instead of 3